### PR TITLE
Add 'make prepare-tools' for offline builds

### PR DESCRIPTION
**Pull request by typeless**
Add a new make target `make prepare-tools` that install all the needed tools for us.
Not sure if this is actually better or helpful at all for use cases like #7997.

### CODE REVIEW

**Comment by typeless** [comment]
It seems that the OpenSUSE build service gets the build prerequisites for a package by a declarative description/configuration. So, this PR probably does not help at all. The users of UBS can handle this on their side without help from Gitea.

File: Makefile [hunk 1]
```diff
@@ -471,8 +438,14 @@ pr:
...
 golangci-lint:
-	@hash golangci-lint > /dev/null 2>&1; if [ $$? -ne 0 ]; then \
-		export BINARY="golangci-lint"; \
-		curl -sfL https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | sh -s -- -b $(GOPATH)/bin v1.17.1; \
-	fi
 	golangci-lint run --deadline=3m
+
+prepare-tools:
+		GO111MODULE="on" $(GO) get -u github.com/go-swagger/go-swagger/cmd/swagger@v0.20.1
+		$(GO) get -u github.com/kisielk/errcheck
```
Comment (sapk) [inline Makefile:447]: Can't we activate GO111MODULE="on" also for the others tools?

Comment (typeless) [reply Makefile:447]: Done.

Comment (typeless) [reply Makefile:447]: I think that is okay.

File: Makefile [hunk 2]
```diff
@@ -471,8 +438,14 @@ pr:
 
 .PHONY: golangci-lint
 golangci-lint:
-	@hash golangci-lint > /dev/null 2>&1; if [ $$? -ne 0 ]; then \
```
Comment (sapk) [inline Makefile:474]: Maybe we can keep those checks but call `make prepare-tools` if needed ?

Comment (typeless) [reply Makefile:474]: Do you mean check the tools at the call sites or check them under `prepare-tools`?

**Comment by typeless** [comment]
This is not

**Pull request closed without merging**